manifests/network.pp

```diff
@@ -7,13 +7,31 @@
   $gateway = '0.0.0.0',
   $type = 'dhcp',
   $lan_channel = 1,
+  $interface_type = 'dedicated',
 )
 {
   require ::ipmi
 
   validate_string($ip,$netmask,$gateway,$type)
   validate_integer($lan_channel)
   validate_re($type, '^dhcp$|^static$', 'Network type must be either dhcp or static')
+  validate_re($interface_type, '^dedicated$|^shared$|^failover$', 'Network interface type must be either dedicated, shared, or failover')
+
+  $interface_type_raw_command_code = '0x30 0x70 0x0c'
+  if $interface_type == 'dedicated' {
```

Comment:
This might be better as a case statement:

```puppet
  case $interface_type {
    'dedicated': {
      # code
    }
    'shared': {
      # code
    }
    'failover': {
      # code
    }
    default: {
      fail('Unknown interface_type')
    }
  }
```